### PR TITLE
chore: create models manager

### PR DESCRIPTION
**Pull request by arnaud-moncel**
## Pull Request checklist:

- [X] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [X] Test manually the implemented changes
- [X] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -77,7 +69,7 @@ function loadCollections(collectionsDir) {
 
 async function buildSchema() {
   const { lianaOptions, Implementation } = configStore;
-  const models = getModels();
+  const models = modelsManager.getModelArray();
```
Comment (jeffladiray) [inline src/index.js:72]: `getModelArray`seems weird IMO.

If I understand correctly, `modelsManager.getModels()` reads `models` in an object, and `getModelArray()`converts it to an array ?

File: test/services/models-manager.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+      const spy = jest.spyOn(modelsManager, '_generateModelList');
+
+      modelsManager.getModels();
+      expect(spy).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('getModelArray', () => {
+    it('should return models as array', () => {
```
Comment (jeffladiray) [inline test/services/models-manager.test.js:46]:
```suggestion
    it('should return models as an array', () => {
```

File: test/services/models-manager.test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            name: 'model2',
+          },
+        })),
+        getModelName: jest.fn((model) => model.name),
+      },
+    };
+
+    const modelsManager = new ModelsManager({ configStore });
+
+    it('should return model lists', () => {
```
Comment (jeffladiray) [inline test/services/models-manager.test.js:21]:
```suggestion
    it('should return the model lists as an object using model name as a key', () => {
```

(Had hard time figuring this out, maybe the label of the test was ok, but a small comment on the difference between `getModels` and `getModelArray`would be nice)

File: src/services/models-manager.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    if (!this.models) this._generateModelList();
+    return this.models;
+  }
+
+  getModelArray() {
+    return _.values(this.getModels());
+  }
+
+  _generateModelList() {
+    const models = this.configStore.Implementation.getModels();
```
Comment (jeffladiray) [inline src/services/models-manager.js:19]: If I'm right, this line is used to be compliant with the current `forest-express` version, and the goal of the following PR is to remove `Implementation` here ?

**Review by jeffladiray — COMMENTED** [review]
Nice work 🥇 

I added a few questions and suggestions to be sure that I fully grasp the goal here (Since you split the previous PR in 2). Other than that, I still need to do a few more tests.

File: src/index.js (hunk 1 above)
Comment (arnaud-moncel) [reply src/index.js:72]: Exactly ;) the only place where we want array is here !

File: src/services/models-manager.js (hunk 4 above)
Comment (arnaud-moncel) [reply src/services/models-manager.js:19]: Yes exactly 👍

**Review by jeffladiray — APPROVED** [review]
LGTM

**Pull request merged**